### PR TITLE
[flutter_tools] remove most globals from build system and tests

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove globals and testbed to simplify test cases, using the existing environment configuration. This is some pre-factoring work to make landing https://github.com/flutter/flutter/pull/63610 easier

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/build_system/targets/android_test.dart [hunk 1]
```diff
@@ -2,83 +2,98 @@
...
+      .file(artifacts.getArtifactPath(Artifact.vmSnapshotData, mode: BuildMode.debug))
+      .createSync(recursive: true);
+    fileSystem
+      .file(artifacts.getArtifactPath(Artifact.isolateSnapshotData, mode: BuildMode.debug))
+      .createSync(recursive: true);
+    // finalsz Directory hostDirectory = fileSystem.currentDirectory
+    //   .childDirectory(getNameForHostPlatform(getCurrentHostPlatform()))
+    //   ..createSync(recursive: true);
+    // hostDirectory.childFile('vm_isolate_snapshot.bin').createSync();
+    // hostDirectory.childFile('isolate_snapshot.bin').createSync();
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/build_system/targets/android_test.dart:67]: Remove? Uncomment?

File: packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart [hunk 2]
```diff
@@ -108,18 +106,22 @@ void main() {
     await expectLater(() => const KernelSnapshot().build(androidEnvironment),
       throwsA(isA<Exception>()));
     expect(processManager.hasRemainingExpectations, false);
-  }));
+  });
 
-  test('KernelSnapshot does not use track widget creation on profile builds', () => testbed.run(() async {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart:113]: How do we still have these just `test`s?

**Review by jmagman — APPROVED** [review]

File: packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart:113]: testbed experiments that didn't pan out.

File: packages/flutter_tools/test/general.shard/build_system/targets/android_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_system/targets/android_test.dart]: Fixed!

**Pull request merged**